content/client-lib-development-guide/features.textile

```diff
@@ -147,18 +147,21 @@ h3(#rest-auth). Auth
+** @(RSA10k)@ If the @AuthOption@ argument's @queryTime@ attribute is true, it will obtain the server time once and persist the offset from the local clock. All future token requests generated directly or indirectly via a call to @authorize@ will not obtain the server time, but instead use the local clock offset to calculate the server time. A client MAY discard the cached local clock offset in situations in which it may have been invalidated, such as if there is a local change to the date, time, or timezone, of the client device
 ** @(RSA10e)@ Adheres to the implementation of @requestToken@ when issuing new tokens
 ** @(RSA10f)@ Returns a @TokenDetails@ object that contains the token string + token metadata
-** @(RSA10g)@ Stores the @AuthOptions@ and @TokenParams@ arguments as defaults for subsequent authorisations with the exception of the attributes @TokenParams#timestamp@, @AuthOptions#queryTime@ and @AuthOptions#force@
+** @(RSA10g)@ Stores the @AuthOptions@ and @TokenParams@ arguments as defaults for subsequent authorisations with the exception of the attributes @TokenParams#timestamp@ and @AuthOptions#queryTime@
 ** @(RSA10h)@ Will use the value from @Auth#clientId@ by default, if not @null@
 ** @(RSA10i)@ Adheres to all requirements in @RSA8@ relating to @TokenParams@, @authCallback@ and @authUrl@
+** @(RSA10l)@ Has an alias method @RestClient#authorise@ and @RealtimeClient#authorise@ that will log a deprecation warning stating that this alias method will be removed in @v1.0@ and the user should instead use @authorize@
+* @(RSA13)@ @Auth#update@ function:
+** @(RSA13a)@ Instructs the library to update the default authentication details in @TokenParams@ and @AuthOptions@ used for all future authentication requests
+** @(RSA13b)@ Method signature is @update(TokenParams, AuthOptions)@. @TokenParams@ and @AuthOptions@ are optional. When the arguments are present, even if empty, the @TokenParams@ and @AuthOptions@ supersede any previously client library configured @TokenParams@ and @AuthOptions@. For example, if a client is initialised with @TokenParams#ttl@ configured with a custom value, and a @TokenParams@ object is passed in as an argument to @#update@ with a @null@ value for @ttl@, then the @ttl@ used for every subsequent authorisation will be @null@
+** @(RSA13c)@ If a previous token does not exist or the token has expired, then a new token will be created immediately, and in the case of a realtime client, the client will re-authenticate, see "RTC8":#RTC8. Please note that a buffer of 15s for token expiry is recommended to avoid race conditions where the token is valid at the time of the request, but invalid when it reaches the server. Therefore, we recommend that a token is considered expired 15s before the time field @expires@
```

Comment:
BTW, just thinking about this. For js, are you imagining this as a function that takes a callback?

I mean, if it could trigger the creation of a new token per RSA13c, it makes sense to have a callback for when that's done. But if it doesn't, then do we callback with null? or with whatever token the lib has at the moment? If the latter, it's unintuitive that a fn called 'update' could callback with a token that doesn't match the tokenParams that were passed in to it. But if it always forces a new token, it's the same as authorize...

To put it another way, what are you thinking of as the usecase for this?